pkg/inference/scheduling/loader.go

```diff
@@ -455,13 +470,19 @@ func (l *loader) release(runner *runner) {
 	// Decrement the runner's reference count.
 	l.references[slot] -= 1
 
-	// If the runner's reference count is now zero, then record now as its idle
-	// start time and signal the idle checker.
+	// If the runner's reference count is now zero, then check if it is still
+	// active, and record now as its idle start time and signal the idle
+	// checker.
 	if l.references[slot] == 0 {
-		l.timestamps[slot] = time.Now()
 		select {
-		case l.idleCheck <- struct{}{}:
+		case <-runner.done:
+			l.evictRunner(runner.backend.Name(), runner.model)
```

Comment:
If we want to get REALLY pedantic, we could also add an optional `mode` specification to `evictRunner` so that eviction of a defunct runner on one mode doesn't force early eviction of an idle runner in another mode.  Most (all?) models aren't used bimodally though, so maybe it's a non-issue.

Reply:
Makes the semantics clearer, I like it :) done.